src/sync-pending-events.ts

```diff
@@ -132,7 +135,7 @@ export class EventTracker {
     let latestBlockNumber = await getLatestBlockNumber(this.source);
     let trueLatestBlockNumber = await getValidBlockNumber(this.source, lastCheckedBlockNumber, latestBlockNumber);
     if (trueLatestBlockNumber) {
-      latestBlockNumber = trueLatestBlockNumber;
+      latestBlockNumber = trueLatestBlockNumber - this.bufferBlocks;
     }else {
       latestBlockNumber = lastCheckedBlockNumber;
     }
```

Comment:
Here, as we had added a new logic to "- this.bufferBlocks",
So this piece of codes from 135 to 141 need to be a function and you can add unit test for this function.
In the unit tests you need cover all edge cases like trueLatestBlockNumber is less than bufferBlocks